packages/babel-helper-optimise-call-expression/test/index.js

```diff
@@ -0,0 +1,103 @@
+    expect(transformInput("a.b(...arguments)")).toMatchInlineSnapshot(
+      `"a.b.apply(a, arguments)"`,
+    );
+    expect(transformInput("a[b](...arguments)")).toMatchInlineSnapshot(
+      `"a[b].apply(a, arguments)"`,
+    );
+    expect(transformInput("a.b?.(...arguments)")).toMatchInlineSnapshot(
+      `"a.b?.apply(a, arguments)"`,
+    );
+    expect(transformInput("a[b]?.(...arguments)")).toMatchInlineSnapshot(
+      `"a[b]?.apply(a, arguments)"`,
+    );
+
+    expect(transformInput("a.b(...args)")).toMatchInlineSnapshot(
+      `"a.b.call(a, ...args)"`,
+    );
+    expect(transformInput("a[b](...args)")).toMatchInlineSnapshot(
+      `"a[b].call(a, ...args)"`,
+    );
+    expect(transformInput("a.b?.(...args)")).toMatchInlineSnapshot(
+      `"a.b?.call(a, ...args)"`,
+    );
+    expect(transformInput("a[b]?.(...args)")).toMatchInlineSnapshot(
+      `"a[b]?.call(a, ...args)"`,
+    );
+
+    expect(transformInput("a.b(arg1, arg2)")).toMatchInlineSnapshot(
+      `"a.b.call(a, arg1, arg2)"`,
+    );
+    expect(transformInput("a[b](arg1, arg2)")).toMatchInlineSnapshot(
+      `"a[b].call(a, arg1, arg2)"`,
+    );
+    expect(transformInput("a.b?.(arg1, arg2)")).toMatchInlineSnapshot(
+      `"a.b?.call(a, arg1, arg2)"`,
+    );
+    expect(transformInput("a[b]?.(arg1, arg2)")).toMatchInlineSnapshot(
+      `"a[b]?.call(a, arg1, arg2)"`,
+    );
+  });
+
+  test("optimizeCallExpression should work when thisNode is provided", () => {
+    expect(transformInput("a.b(...arguments)", "c")).toMatchInlineSnapshot(
+      `"a.b.apply(c, arguments)"`,
+    );
+    expect(transformInput("a[b](...arguments)", "c")).toMatchInlineSnapshot(
+      `"a[b].apply(c, arguments)"`,
+    );
+    expect(transformInput("a.b?.(...arguments)", "c")).toMatchInlineSnapshot(
+      `"a.b?.apply(c, arguments)"`,
+    );
+    expect(transformInput("a[b]?.(...arguments)", "c")).toMatchInlineSnapshot(
+      `"a[b]?.apply(c, arguments)"`,
+    );
+
+    expect(transformInput("a.b(...args)", "c")).toMatchInlineSnapshot(
+      `"a.b.call(c, ...args)"`,
+    );
+    expect(transformInput("a[b](...args)", "c")).toMatchInlineSnapshot(
+      `"a[b].call(c, ...args)"`,
+    );
+    expect(transformInput("a.b?.(...args)", "c")).toMatchInlineSnapshot(
+      `"a.b?.call(c, ...args)"`,
+    );
+    expect(transformInput("a[b]?.(...args)", "c")).toMatchInlineSnapshot(
+      `"a[b]?.call(c, ...args)"`,
+    );
+
+    expect(transformInput("a.b(arg1, arg2)", "c")).toMatchInlineSnapshot(
+      `"a.b.call(c, arg1, arg2)"`,
+    );
+    expect(transformInput("a[b](arg1, arg2)", "c")).toMatchInlineSnapshot(
+      `"a[b].call(c, arg1, arg2)"`,
+    );
+    expect(transformInput("a.b?.(arg1, arg2)", "c")).toMatchInlineSnapshot(
+      `"a.b?.call(c, arg1, arg2)"`,
+    );
+    expect(transformInput("a[b]?.(arg1, arg2)", "c")).toMatchInlineSnapshot(
+      `"a[b]?.call(c, arg1, arg2)"`,
```

Comment:
None of these tests uses a private property, could you add one?

Reply:
Addressed in b783079.